mapper/orm/src/main/java/org/hibernate/search/mapper/orm/loading/impl/AbstractHibernateOrmLoadingStrategy.java

```diff
@@ -37,14 +41,58 @@
 public abstract class AbstractHibernateOrmLoadingStrategy<E, I>
 		implements HibernateOrmEntityLoadingStrategy<E, I> {
 
+	enum GroupingAllowed {
+		NEVER {
+			@Override
+			public boolean allowed(boolean hasNonIndexedConcreteSubtypes) {
+				return false;
+			}
+		},
+		ALWAYS {
+			@Override
+			public boolean allowed(boolean hasNonIndexedConcreteSubtypes) {
+				return true;
+			}
+		},
+		ONLY_FOR_NO_NON_INDEXED_CONCRETE_SUBTYPES {
+			@Override
+			public boolean allowed(boolean hasNonIndexedConcreteSubtypes) {
+				return !hasNonIndexedConcreteSubtypes;
+			}
+		};
+
+		public static GroupingAllowed determine(PersistentClass persistentClass) {
+			if ( isFromInheritanceType( persistentClass, JoinedSubclass.class ) ) {
```

Comment:
This should also address the root classes 🙈

Reply:
"Should" as in "you added a test and you're sure"?

Reply:
Yup, there are tests that try indexing from the root as well as from entities deeper in the hierarchy. We definitely can have more tests, but since adding more of these hierarchies to add new variations of possible configurations make the test grow too-big-to-read (and potentially to update the regexps) I'd say let's keep it at what we have and add in the future if some new case comes up. 🙈 😃